docs/proposals/old-timer.md

```diff
@@ -0,0 +1,146 @@
+list of aggregations to retrieve multiple at once.
+
+Where `{prefix}` is either empty (cluster-level), `/namespaces/{namespace}`
+(namespace-level), `/namespaces/{namespace}/pods/{pod-name}` (pod-level),
+`/namespaces/{namespace}/pod-list/{pod-list}` (multi-pod-level), or
+`/namespaces/{namespace}/pods/{pod-name}/containers/{container-name}`
+(container-level).
+
+In addition, when `{prefix}` is not empty, there will be a url of the form:
+`/api/v1/historical/{prefix-without-final-element}` which allows fetching the
+list of available nodes/namespaces/pods/containers.
+
+The `start` and `end` parameters are defined the same way as for the model.
+The `bucket` (bucket duration) parameter is a number followed by any of the
+following suffixes:
+
+- `ms`: milliseconds
+- `s`: seconds
+- `m`: minutes
+- `h`: hours
+- `d`: days
+
+### Functionality
+
+When Oldtimer receives a request at one of the given URLs, it will compose a
+query to the configured metrics sink, execute that query, and return the
+results.  The return format for normal requests will be the same as that
+returned by the Heapster model.
+
+In the case of aggregations, the normal `MetricsResult` and `MetricsResultList`
+are wrapped in order to differentiate between different aggregations.  Each
+metric point represents one bucket (if no buckets are requested, only one point
+is returned).  The timestamp in the case of aggregations is the timestamp of
+the start of that bucket.
+
+```go
+type MetricAggregationResult struct {
+    Average *MetricResult
+    Maximum *MetricResult
+    Minimum *MetricResult
+    Median *MetricResult
+    Count *MetricResult
+    Percentiles map[uint64]MetricResult
+}
+
+type MetricListAggregationResult struct {
+    Average *MetricResultList
+    Maximum *MetricResultList
+    Minimum *MetricResultList
+    Median *MetricResultList
+    Count *MetricResultList
+    Percentiles map[uint64]MetricResultList
+}
+```
+
+### Aggregations
+
+Several different aggregations will be supported.  Aggregations should be
+performed in the metrics sink.  If more aggregations later become supported
+across all metrics sinks, the list can be expanded (and the API version
+should probably be bumped, since the supported aggregations should be part of
+the API).
+
+- Average (arithmetic mean): `/{metric-name}/average`
+- Maximum: `/{metric-name}/max`
+- Minimum: `/{metric-name}/min`
+- Percentile: `/{metric-name}/{number}-perc`
+- Median: `/{metric-name}/median`
+- Count: `/{metric-name}/count`
+
+## Scaling and Performance Considerations
+
+Since Oldtimer itself does not store any data, it should be fairly easy to
+deploy multiple replicas of Oldtimer.  The metrics sinks themselves should
+already have clustering support, and thus can be scaled as well.  Since
+Oldtimer queries the metrics sinks themselves, response latency should
+depend mainly on how quickly the sinks can respond to queries.
+
+## Open Questions
+
```

Comment:
The goal is to have a set of features that all the current metrics sinks support, and which seem likely for any future metrics sinks to support.